src/main/java/com/google/cloud/tools/gradle/appengine/appyaml/AppEngineAppYamlPlugin.java

```diff
@@ -89,11 +90,11 @@ private void configureExtensions() {
           // we can only set the default location of "archive" after project evaluation (callback)
           if (stageExtension.getArtifact() == null) {
             if (project.getPlugins().hasPlugin(WarPlugin.class)) {
-              War war = (War) project.getProperties().get("war");
-              stageExtension.setArtifact(war.getArchivePath());
+              stageExtension.setArtifact(
+                  tasks.withType(War.class).named("war").map(war -> war.getArchivePath()));
             } else if (project.getPlugins().hasPlugin(JavaPlugin.class)) {
-              Jar jar = (Jar) project.getProperties().get("jar");
-              stageExtension.setArtifact(jar.getArchivePath());
+              stageExtension.setArtifact(
+                  tasks.withType(Jar.class).named("jar").map(jar -> jar.getArchivePath()));
```

Comment:
no need to create a filtered task collection:
```suggestion
                  tasks.named("jar", Jar.class).map(jar -> jar.getArchivePath()));
```